pyresample/future/spherical/point.py

```diff
@@ -39,13 +39,32 @@ def from_degrees(cls, lon, lat):
         """Create SPoint from lon/lat coordinates in degrees."""
         return cls(np.deg2rad(lon), np.deg2rad(lat))
 
+    def is_pole(self):
+        """Test if the point is on a pole."""
+        if self.lat in [-np.pi / 2, np.pi / 2]:
+            return True
+        return False
+
+    def is_on_equator(self):
+        """Test if the point is on the equator."""
+        if self.lat == 0:
+            return True
+        return False
+
     def __str__(self):
         """Get simplified representation of lon/lat arrays in radians."""
         return str((float(self.lon), float(self.lat)))
 
     def __repr__(self):
         """Get simplified representation of lon/lat arrays in radians."""
-        return str((float(self.lon), float(self.lat)))
+        coords = str((float(self.lon), float(self.lat)))
+        return f"{self.__class__.__name__}: {coords}"
+
+    def __eq__(self, other):
+        """Check equality."""
+        if self.lat == other.lat and self.is_pole():
+            return True
+        return np.allclose((self.lon, self.lat), (other.lon, other.lat))
```

Comment:
I'm tempted to want this to be like the below, but maybe how you have it is more clear:

```suggestion
        return (self.lat == other.lat and self.is_pole()) or np.allclose((self.lon, self.lat), (other.lon, other.lat))
```